server/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1838,20 +1840,41 @@ public static String createEmptyTranslog(final Path location, final long initial
 
     static String createEmptyTranslog(Path location, long initialGlobalCheckpoint, ShardId shardId,
                                       ChannelFactory channelFactory, long primaryTerm) throws IOException {
+        return createEmptyTranslog(location, shardId, initialGlobalCheckpoint, primaryTerm, null, channelFactory);
+    }
+
+    public static String createEmptyTranslog(final Path location,
```

Comment:
Suggest adding a big warning in a comment here saying how dangerous it is to specify the translog UUID and that it should only be used for shards that will see no indexing.

I'm also idly wondering about how hard it would be to make this `Translog` read-only when it's not been created with a fresh UUID. Probably too hard to be worth doing, but thought I'd mention it anyway.

Reply:
> Suggest adding a big warning in a comment here saying how dangerous it is to specify the translog UUID and that it should only be used for shards that will see no indexing.

Sure, I added some doc in 557d924f02ea84cbd084734a132277706bfead62

> I'm also idly wondering about how hard it would be to make this `Translog` read-only when it's not been created with a fresh UUID. Probably too hard to be worth doing, but thought I'd mention it anyway.

I find the idea interesting but I'm not sure if it worths it; I'd prefer to not create translogs at all if they were not to be used.